usergroups.go

```diff
@@ -0,0 +1,172 @@
+package slack
+
+import (
+	"errors"
+	"net/url"
+	"strings"
+)
+
+// UserGroup contains all the information of a user group
+type UserGroup struct {
+	ID          string `json:"id"`
+	TeamID      string `json:"team_id"`
+	IsUserGroup bool   `json:"is_usergroup"`
+	Name        string `json:"name"`
+	Description string `json:"description"`
+	Handle      string `json:"handle"`
+	IsExternal  bool   `json:"is_external"`
+	DateCreate  int    `json:"date_create"`
+	DateUpdate  int    `json:"date_update"`
+	DateDelete  int    `json:"date_delete"`
+	AutoType    string `json:"auto_type"`
+	CreatedBy   string `json:"created_by"`
+	UpdatedBy   string `json:"updated_by"`
+	DeletedBy   string `json:"deleted_by"`
```

Comment:
`deleted_by` can be null or an ID but not an empty string as well, it seemed like that would only be necessary if distinguishing between all three?